File: services/gateway/app/api/incidents.py
from fastapi import APIRouter


router = APIRouter()


_DEMO_INCIDENTS = [
    {
        "id": "inc-001",
        "title": "payment-api OOMKilled",
        "service": "payment-api",
        "severity": "critical",
        "status": "investigating",
        "agent": "investigation-agent",
    },
    {
        "id": "inc-002",
        "title": "gateway latency spike",
        "service": "gateway",
        "severity": "warning",
        "status": "analyzing",
        "agent": "investigation-agent",
    },
]


@router.get("/incidents")
def list_incidents():
    return {
        "items": _DEMO_INCIDENTS,
        "total": len(_DEMO_INCIDENTS),
    }


@router.get("/incidents/{incident_id}")
def get_incident(incident_id: str):
    for incident in _DEMO_INCIDENTS:
        if incident["id"] == incident_id:
            return incident

    return {
        "id": incident_id,
        "status": "unknown",
    }

File: web/lib/api/incidents.ts
export interface Incident {
  id: string;
  title: string;
  service: string;
  severity: string;
  status: string;
  agent?: string;
}

const API_BASE = process.env.NEXT_PUBLIC_API_BASE ?? "http://localhost:8000";

export async function listIncidents(): Promise<Incident[]> {
  const response = await fetch(`${API_BASE}/api/incidents`, {
    cache: "no-store",
  });

  if (!response.ok) {
    throw new Error("failed to fetch incidents");
  }

  const data = await response.json();
  return data.items ?? [];
}
